rfcs/0004-enhanced-limits-and-capabilities.md

```diff
@@ -0,0 +1,655 @@
++              args: ["/C", "{{Task.File.<implicitly generated file>}}", "--additional-argument"]
++      embeddedFiles:
++          - name: <implicitly generated file>
++            filename: "<implicitly generated file>.bat"
++            type: TEXT
++            data:
++                # batch code here
++      ```
++
++   4. *powershell* - Implicitly creates a Powershell embedded file,
++   `command` becomes `powershell`, and `args` get prefixed with
++   `["-File", "{{Task.File.<implicitly generated file>}}"]`.
++   The file extension is `.ps1`.
++
++      ```yaml
++      actions:
++        onRun:
++            args: ["--additional-argument"] # optional
++            powershell:
++                # powershell code here
++
++      ### syntax sugar equivalent to:
++
++      actions:
++          onRun:
++              command: powershell
++              args: ["-File", "{{Task.File.<implicitly generated file>}}", "--additional-argument"]
++      embeddedFiles:
++          - name: <implicitly generated file>
++            filename: "<implicitly generated file>.ps1"
++            type: TEXT
++            data:
++                # Powershell code here
++      ```
++
++   5. *node* - Implicitly creates a JavaScript embedded file,
++   `command` becomes `node`, and `args` get prefixed with the implicitly
++   generated file. The file extension is `.js`.
++
++      ```yaml
++      actions:
++        onRun:
++            args: ["--additional-argument"] # optional
++            node:
++                # JavaScript code here
++
++      ### syntax sugar equivalent to:
++
++      actions:
++          onRun:
++              command: node
++              args: ["{{Task.File.<implicitly generated file>}}", "--additional-argument"]
++      embeddedFiles:
++          - name: <implicitly generated file>
++            filename: "<implicitly generated file>.js"
++            type: TEXT
++            data:
++                # JavaScript code here
++      ```
++
+ The host uses the return code of the **command** run to determine success or failure of the Action. A zero exit code
+ indicates success, and any non-zero exit code indicates failure. A timeout also indicates failure.
+````
+
+## Prior Art
+
+### Parameter Limits
+
+* **Deadline 10**: No upper limit, most plugins have ~20 options. With outliers
+such as VRED with 61 options and 3dsMax with 2185 options.
+
+### Script Interpreter Support
+
+* **GitHub Actions**: Provides `shell` property with predefined options
+
+## Rejected Ideas
+
+### Multiple Actions in onRun
+
+The original discussion included a proposal to generalize the `onRun` action
+property in `<StepActions>` to accept either a single `<Action>` as currently
+defined, or a list of `<Action>` objects. The goal was to allow a list of
+sequential actions in the `onRun` of a task, with each command having separate
+timeout and cancelation behavior.
+
+This feature was rejected for inclusion in this RFC because it introduces
```

Comment:
Since the idea is rejected from the RFC, not in general, I'd suggest to not mention it at all, and rather create a new discussion topic dedicated to it and comment on the existing one that this point is moved there.